packages/grid/x-data-grid-premium/src/hooks/features/cellSelection/useGridCellSelection.ts

```diff
@@ -0,0 +1,363 @@
+          (acc2, column) => {
+            acc2[column.field] = true;
+            return acc2;
+          },
+          {},
+        );
+        return acc;
+      }, {});
+
+      apiRef.current.setCellSelectionModel(newModel);
+    },
+    [apiRef, visibleRows.rows],
+  );
+
+  const getSelectedCellsAsArray = React.useCallback<
+    GridCellSelectionApi['getSelectedCellsAsArray']
+  >(() => {
+    const model = apiRef.current.getCellSelectionModel();
+    const idToIdLookup = gridRowsDataRowIdToIdLookupSelector(apiRef);
+
+    return Object.entries(model).reduce<{ id: GridRowId; field: string }[]>(
+      (acc, [id, fields]) => [
+        ...acc,
+        ...Object.entries(fields).reduce<{ id: GridRowId; field: string }[]>(
+          (acc2, [field, isSelected]) => {
+            return isSelected ? [...acc2, { id: idToIdLookup[id], field }] : acc2;
+          },
+          [],
+        ),
+      ],
+      [],
+    );
+  }, [apiRef]);
+
+  const cellSelectionApi: GridCellSelectionApi = {
+    isCellSelected,
+    getCellSelectionModel,
+    setCellSelectionModel,
+    selectCellRange,
+    getSelectedCellsAsArray,
+  };
+
+  useGridApiMethod(apiRef, cellSelectionApi, 'public');
+
+  const handleCellMouseDown = React.useCallback<GridEventListener<'cellMouseDown'>>(
+    (params) => {
+      lastMouseDownCell.current = { id: params.id, field: params.field };
+      apiRef.current.rootElementRef?.current?.classList.add(
+        gridClasses['root--disableUserSelection'],
+      );
+    },
+    [apiRef],
+  );
+
+  const handleCellMouseUp = React.useCallback<GridEventListener<'cellMouseDown'>>(() => {
+    lastMouseDownCell.current = null;
+    apiRef.current.rootElementRef?.current?.classList.remove(
+      gridClasses['root--disableUserSelection'],
+    );
+  }, [apiRef]);
+
+  const handleCellMouseOver = React.useCallback<GridEventListener<'cellMouseOver'>>(
+    (params) => {
+      if (!lastMouseDownCell.current) {
+        return;
+      }
+
+      const { id, field } = params;
+      apiRef.current.selectCellRange(lastMouseDownCell.current, { id, field });
+    },
+    [apiRef],
+  );
+
+  const handleCellClick = useEventCallback<
+    [GridEventLookup['cellClick']['params'], GridEventLookup['cellClick']['event']],
+    void
+  >((params, event) => {
+    const { id, field } = params;
+
+    if (params.field === GRID_CHECKBOX_SELECTION_COL_DEF.field) {
+      return;
+    }
+
+    if (params.field === GRID_DETAIL_PANEL_TOGGLE_FIELD) {
+      return;
+    }
+
+    const column = apiRef.current.getColumn(params.field);
+    if (column.type === GRID_ACTIONS_COLUMN_TYPE) {
+      return;
+    }
+
+    if (params.rowNode.type === 'pinnedRow') {
+      return;
+    }
+
+    if (event.shiftKey && lastClickedCell.current) {
+      apiRef.current.selectCellRange(lastClickedCell.current, { id, field });
+      lastClickedCell.current = { id, field };
+      return;
+    }
+
+    lastClickedCell.current = { id, field };
+
+    if (event.ctrlKey) {
```

Comment:
For macOS?

```suggestion
    if (event.ctrlKey || event.metaKey) {
```

I tested it, and can now use Cmd like on Google Sheet.